File: Leetcode/328-Odd-Even-Linked-List/OddEvenLinkedList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def oddEvenList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None:
            return head

        o = head
        e = head.next
        eHead = head.next
        while e != None and e.next != None:
            o.next = e.next
            e.next = e.next.next
            o = o.next
            e = e.next

        o.next = eHead
        return head


    

File: Leetcode/328-Odd-Even-Linked-List/OddEvenLinkedList.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func oddEvenList(head *ListNode) *ListNode {
	if head == nil {
		return head
	}

	o := head
	e := head.Next
	eHead := head.Next
	for e != nil && e.Next != nil {
		o.Next = e.Next
		e.Next = e.Next.Next
		o = o.Next
		e = e.Next
	}
	o.Next = eHead
	return head
}
